internal/campaigns/repo_fetcher.go

```diff
@@ -47,92 +76,82 @@ type RepoZip interface {
 	Path() string
 }
 
+var _ RepoZip = &repoZip{}
+
 // repoZip is the concrete implementation of the RepoZip interface used outside
 // of tests.
 type repoZip struct {
-	path    string
-	fetcher *repoFetcher
+	mu sync.Mutex
+
+	deleteOnClose bool
+	path          string
+	repo          *graphql.Repository
 
-	mu         sync.Mutex
+	client api.Client
+
+	// references is the number of *active* tasks that currently use the archive.
 	references int
+	// marks is the number of tasks that *will* make use of the archive.
+	marks int
```

Comment:
Not a blocker, but `marks` felt very unintuitive to me, maybe something like `checkouts` or so would be clearer?

Reply:
Yes! Sorry, that is a leftover.